Using-GTSAM-EXPORT.md

```diff
@@ -0,0 +1,41 @@
+
+## But Why?
+To me, the behavior of GTSAM_EXPORT was very baffling.  However, I believe it can be explained by understanding the rules that MSVC operates under.  
+
+First, we need to understand exactly what `GTSAM_EXPORT` does.  When you use `GTSAM_EXPORT` on a function (or class) definition, it actually does two things.  First, when you are building the library, it turns into a "dllexport" command, telling the compiler that this function should go into the DLL and be visible externally.  When you pull in that same header for a different library, it switches to a "dllimport" command, telling the compiler it should find this function in a DLL somewhere.  This leads to the first rule the compiler uses.  (Note that I say "compiler rules" when the rules may actually be in the linker, but I am conflating the two terms here when I speak of the "compiler rules".) 
+
+***Compiler Rule #1*** If a `dllimport` command is used in defining a function or class, that function or class _must_ be found in a DLL.
+
+Rule #1 doesn't seem very bad, until you combine it with rule #2
+
+***Compiler Rule #2*** Anything defined in a header file is not included in a DLL.
+
+When these two rules are combined, you get some very confusing results.  For example, a class which is completely defined in a header (e.g. LieMatrix) cannot use `GTSAM_EXPORT` in its definition.  If LieMatrix is defined with `GTSAM_EXPORT`, then the compiler _must_ find LieMatrix in a DLL.  Because LieMatrix is a header-only class, however, it can't find it, leading to a very confusing "I can't find this symbol" type of error.  Note that the linker says it can't find the symbol even though the compiler found the header file that completely defines the class.
+
+This leads to another compiler rule that finishes explaining the usage rules proposed above.
+
+***Compiler Rule #3*** If a class is defined with `GTSAM_EXPORT`, then any classes it inherits from must be definable with a dllexport command as well.
+
+One of the most immediate results of this compiler rule is that any class that inherits from an Eigen class _cannot_ be defined with `GTSAM_EXPORT`.  Because Eigen is a header-only class, compiler rule #2 prohibits any Eigen class from being defined with a "dllexport" command.  Therefore, no Eigen inherited classes can use `GTSAM_EXPORT` either.  (Note that individual functions with an inherited class can be "exported."  Just not the entire class.)
```

Comment:
It's a dirty solution, but if that's the unique way you found to solve the linking problems, let is be!  

Just let's iterate once over this text, and in the meanwhile give me some time to check it on a local machine before merging. 

Awesome work!

Reply:
Thanks.  Please note that there are still compiler errors in several of the "check" classes that I am slowly working my way through.  The matlab wrappers seem to work, which is my main concern, but while I am at it, I would like to get the "check" classes all working as well.

Also, @dellaert I compiled the libraries with the TYPEDEF_POINT_AS_VECTOR option enabled and did not see any new errors pop up.